conda_smithy/templates/github-actions.yml.tmpl

```diff
@@ -185,47 +192,72 @@ jobs:
 {%- for secret in secrets %}
         {{ secret }}: {% raw %}${{{% endraw %} secrets.{{ secret }} {% raw %}}}{% endraw %}
 {%- endfor %}
-      if: matrix.os == 'windows'
-
 {%- if github_actions.store_build_artifacts %}
+
+    - name: Determine build outcome
+      # this is to merge the status of the linux/osx/win builds into
+      # something we can easily reuse during artefact generation
+      id: determine-status
+      shell: bash
+      env:
+        OS: {% raw %}${{ matrix.os }}{% endraw %}
+      run: |
+        if [[ "$OS" == "ubuntu" ]]; then
+          STATUS={% raw %}${{ steps.build-linux.outcome }}{% endraw %}
+        elif [[ "$OS" == "macos" ]]; then
+          STATUS={% raw %}${{ steps.build-macos.outcome }}{% endraw %}
+        elif [[ "$OS" == "windows" ]]; then
+          STATUS={% raw %}${{ steps.build-windows.outcome }}{% endraw %}
+        fi
+        if [ -z "$STATUS" ]; then
+          # steps that never ran will have empty status
+          STATUS="cancelled"
+        fi
+        echo "status=$STATUS" >> $GITHUB_OUTPUT
+
     - name: Prepare conda build artifacts
+      continue-on-error: true
       id: prepare-artifacts
       shell: bash
-      if: {% raw %}${{ always() }}{% endraw %}
+      # we do not want to trigger artefact creation if the build was cancelled
+      if: {% raw %}${{ steps.determine-status.outputs.status != 'cancelled' }}{% endraw %}
```

Comment:
OK, I didn't know GHA's inner workings well enough to understand that the `always()` here is necessary, as otherwise GHA might decide to skip steps after failure. Pushed a fix-up in 4c30cf1d9c446ce77553c298dc652418e6595a20